src/ucs/datastruct/ptr_array.h

```diff
@@ -57,61 +73,83 @@ typedef struct ucs_ptr_array {
 /**
  * Initialize the array.
  *
- * @param init_placeholder   Default placeholder value.
+ * @param [in] ptr_array          Pointer to a ptr array.
+ * @param [in] name               The name of the ptr array.
  */
-void ucs_ptr_array_init(ucs_ptr_array_t *ptr_array, uint32_t init_placeholder,
-                        const char *name);
+void ucs_ptr_array_init(ucs_ptr_array_t *ptr_array, const char *name);
 
 
 /**
  * Cleanup the array.
- * All values should already be removed from it.
+ *
+ * @param ptr_array  Pointer to a ptr array.
+ *
+ * @note All values should already be removed from it.
  */
 void ucs_ptr_array_cleanup(ucs_ptr_array_t *ptr_array);
 
 
 /**
  * Insert a pointer to the array.
  *
- * @param value        Pointer to insert. Must be 8-byte aligned.
- * @param placeholder  Filled with placeholder value.
- * @return             The index to which the value was inserted.
+ * @param [in] ptr_array     Pointer to a ptr array.
+ * @param [in] value         Pointer to insert. Must be 8-byte aligned.
+ *
+ * @return The index to which the value was inserted.
  *
  * Complexity: amortized O(1)
  *
- * Note: The array will grow if needed.
+ * @note The array will grow if needed.
  */
-unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value,
-                              uint32_t *placeholder_p);
+unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value);
 
 
 /**
  * Remove a pointer from the array.
  *
- * @param index        Index to remove from.
- * @param placeholder  Value to put in the free slot.
+ * @param [in] ptr_array      Pointer to a ptr array.
+ * @param [in] element_index  Index to remove from.
  *
  * Complexity: O(1)
  */
-void ucs_ptr_array_remove(ucs_ptr_array_t *ptr_array, unsigned index,
-                          uint32_t placeholder);
+void ucs_ptr_array_remove(ucs_ptr_array_t *ptr_array, unsigned element_index);
 
 
 /**
  * Replace pointer in the array
- * @param  index    index of slot
- * @param  new_val  value to put into slot given by index
- * @return old value of the slot
+ *
+ * @param [in] ptr_array      Pointer to a ptr array.
+ * @param [in] element_index  Index of slot
```

Comment:
missing `.` in the end here and below (including  `return`)

Reply:
Fixed.